### ISSUE TITLE
[Accessibility] Focus and label issues with help popups

### ISSUE THREAD

**Issue by PeterSaathoff-Harshfield** [ampproject/amp.dev#4500]
Environment:
Any OS/browser and screen reader (such as VoiceOver, NVDA, or ChromeVox)

Steps to reproduce:
1. Open https://amp-dev-staging.appspot.com/page-experience/
2. Enter a valid URL and press Analyze
3. Press tab repeatedly to move keyboard focus to the question mark/help buttons
4. Press enter to open a popup
5. Continue to press tab to move keyboard focus past the popup

Expected:
The close button and Learn more link in the popups should not be keyboard focusable unless the popup is open.
It should not be possible to move keyboard focus outside the popup without the popup closing.
When the popup opens, keyboard focus should move to the close button, and the question mark button should not be focusable when the popup is open. When the popup closes, focus should return to the question mark button.
The question mark and close buttons should have labels, which are read by screen readers.

Actual:
The close button and Learn more link are keyboard focusable when the popup is not open.
It is possible to move keyboard focus to controls outside the popup when it's open.
When the question mark button is activated (by pressing enter), keyboard focus remains on the button.
The question mark button and close button both have no label.

This popup is functionally a modal dialog [Creating An Accessible Modal Dialog](https://bitsofco.de/accessible-modal-dialog/)

**Comment by caroqliu** [comment]
Having trouble trapping focus after trying the following approaches:
## Prevent Tab key from leaving focus
event.preventDefault() not supported in worker-dom

## Cycle from last element to first and vice versa to trap focus
Element.focus() not supported in worker-dom

## Count focused location among focusable elements
- Start count at 0
- Increase by 1 with `Tab`
- Decrease by 1 with `Shift` + `Tab`
- Exit when index < 0 or >= length of focusable elements 
- event.shiftKey not supported in worker-dom, so no ability to count going backwards (forwards is possible however)

## Hide with CSS :focus-within
- The idea: Visible while focus inside and invisible while focus outside.
- The following rule is paradoxical:

```
not(:focus-within) {
  visibility: hidden;
}
```

This is because so long as the target element is hidden, it is not possible to focus inside it.

**Comment by caroqliu** [comment]
Having trouble trapping focus after trying the following approaches:
## Prevent Tab key from leaving focus
[Sample change](https://github.com/ampproject/amp.dev/commit/9ffc21027f2516309241f6f2c2fe81d2adda9d18) - event.preventDefault() not supported in worker-dom

## Cycle from last element to first and vice versa to trap focus
Element.focus() not supported in worker-dom

## Count focused location among focusable elements
- Start count at 0
- Increase by 1 with `Tab`
- Decrease by 1 with `Shift` + `Tab`
- Exit when index < 0 or >= length of focusable elements 
- event.shiftKey not supported in worker-dom, so no ability to count going backwards (forwards is possible however)

## Hide with CSS :focus-within
- The idea: Visible while focus inside and invisible while focus outside.
- The following rule is paradoxical:

```
not(:focus-within) {
  visibility: hidden;
}
```

This is because so long as the target element is hidden, it is not possible to focus inside it.

## Alternatives 
We could attempt to replace all tooltip implementations with `<amp-lightbox>`

**Comment by caroqliu** [comment]
Chatted with @PeterSaathoff-Harshfield with the following resolution: 
> Now that I read about this more, it seems like it could make more sense for it to be a non-modal dialog. In that case, activating the question mark button would open/close it and focus would not be trapped. The close button would be unnecessary, and the question mark button would need to expose an expanded/collapsed state for screen readers.

While the close button is unnecessary I may still include it -- in that case we would have a toggle question mark and a toggle off close button. Implemented in #4607

**Comment by caroqliu** [comment]
@PeterSaathoff-Harshfield This fix is now available in https://amp-dev-staging.appspot.com/page-experience/, can you please comment once verified?

**Comment by PeterSaathoff-Harshfield** [comment]
Looks good 👍

**Comment by caroqliu** [comment]
Thank you!!
